src/main/java/edu/harvard/iq/dataverse/util/json/JsonPrinter.java

```diff
@@ -373,17 +373,30 @@ public static JsonObjectBuilder json(DatasetVersion dsv) {
     }
 
     public static JsonObjectBuilder json(DatasetVersion dsv, List<String> anonymizedFieldTypeNamesList) {
+        Dataset dataset = dsv.getDataset();
         JsonObjectBuilder bld = jsonObjectBuilder()
-                .add("id", dsv.getId()).add("datasetId", dsv.getDataset().getId())
-                .add("datasetPersistentId", dsv.getDataset().getGlobalId().asString())
-                .add("storageIdentifier", dsv.getDataset().getStorageIdentifier())
+                .add("id", dsv.getId()).add("datasetId", dataset.getId())
+                .add("datasetPersistentId", dataset.getGlobalId().asString())
+                .add("storageIdentifier", dataset.getStorageIdentifier())
```

Comment:
jsonObjectBuilder() is already NullSafeJsonBuilder to it should be safe to add

.add("alternativePersistentId", dataset.getAlternativePersistentIdentifier())

for example.